lib/ansible/modules/cloud/smartos/vmadm.py

```diff
@@ -0,0 +1,612 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2017, Jasper Lievisse Adriaanse <j@jasper.la>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible. If not, see <http://www.gnu.org/licenses/>.
+#
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: vmadm
+short_description: Manage SmartOS virtual machines and zones
+description:
+    - Manage SmartOS virtual machines through vmadm(1M)
+version_added: "2.3"
+author: Jasper Lievisse Adriaanse (@jasperla)
+options:
+    save_payload:
+        required: false
+        description:
+          - Don't delete the JSON payload used to generated a VM.
+    force:
+        required: false
+        description:
+          - Force a particular action (i.e. stop or delete a VM).
+    state:
+        required: true
+        choices: [ present, absent, stopped, restarted ]
+        description:
+          - States for the VM to be in. Please note that C(present), C(stopped) and C(restarted)
+            operate on a VM that is currently provisioned. C(present) means that the VM will be
+            created if it was absent, and that it will be in a running state. C(absent) will
+            shutdown the zone before removing it.
+            C(stopped) means the zone will be created if it doesn't exist already, before shutting
+            it down.
+    alias:
```

Comment:
Ok, I originally named it `alias` as that's the regular idiom. But I'll make alias an alias for name.